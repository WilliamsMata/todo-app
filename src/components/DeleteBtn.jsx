import PropTypes from "prop-types";

export const DeleteBtn = ({ onClick }) => {
  return (
    <button className="btn btn-danger d-flex" onClick={onClick}>
      <img
        style={{ width: 20, aspectRatio: 1 / 1.5 }}
        className="align-self-center justify-self-center"
        src="./delete-alt-svgrepo-com.svg"
        alt="delete"
      />
    </button>
  );
};

DeleteBtn.propTypes = {
  onClick: PropTypes.func,
};
