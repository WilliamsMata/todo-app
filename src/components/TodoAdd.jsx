import { useForm } from "../hooks/useForm";
import PropTypes from "prop-types";

export const TodoAdd = ({ onNewTodo }) => {
  const { description, onInputChange, onResetForm } = useForm({
    description: "",
  });

  const onSubmit = (event) => {
    event.preventDefault();
    if (description.trim().length <= 1) return;
    const newTodo = {
      id: new Date().getTime(),
      description: description.trim(),
      done: false,
    };
    onNewTodo(newTodo);
    onResetForm();
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        className="form-control"
        type="text"
        placeholder="Que hay que hacer?"
        autoComplete="off"
        name="description"
        value={description}
        onChange={onInputChange}
      />

      <button type="submit" className="btn btn-primary mt-2">
        Agregar Tarea
      </button>
    </form>
  );
};

TodoAdd.protoTypes = {
  onNewTodo: PropTypes.func,
};
