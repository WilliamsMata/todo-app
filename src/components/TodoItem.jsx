import { DeleteBtn } from "./DeleteBtn";

export const TodoItem = ({ todo, onDeleteTodo, onToggleTodo }) => {
  return (
    <li className="list-group-item d-flex justify-content-between align-items-center p-0}">
      <span
        className={`align-self-center flex-grow-1 pt-2 pb-2 ${
          todo.done ? "text-decoration-line-through" : ""
        }`}
        style={{ cursor: "pointer" }}
        onClick={() => onToggleTodo(todo.id)}
        aria-label="span"
      >
        {todo.description}
      </span>

      <DeleteBtn onClick={() => onDeleteTodo(todo.id)} />
    </li>
  );
};
