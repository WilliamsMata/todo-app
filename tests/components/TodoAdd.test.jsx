import { fireEvent, render, screen } from "@testing-library/react";
import { TodoAdd } from "../../src/components/TodoAdd";

describe("Pruebas en <TodoAdd />", () => {
  const onNewTodoMock = jest.fn();
  beforeEach(() => jest.clearAllMocks());

  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<TodoAdd />);

    expect(container).toMatchSnapshot();
  });

  test("Se renderiza correctamente un formulario con un input y un botón.", () => {
    render(<TodoAdd />);

    const inputText = screen.getByRole("textbox");
    const submitBtn = screen.getByRole("button");

    expect(inputText).toBeTruthy();
    expect(submitBtn).toBeTruthy();
  });

  test("Debe de llamar el onNewTodo con el objeto de la tarea y luego debe limpiar el textbox", () => {
    render(<TodoAdd onNewTodo={onNewTodoMock} />);
    const inputText = screen.getByRole("textbox");
    const submitBtn = screen.getByRole("button");

    fireEvent.input(inputText, { target: { value: "Nueva Tarea" } });
    fireEvent.click(submitBtn);

    expect(onNewTodoMock).toHaveBeenCalledWith({
      id: expect.any(Number),
      description: "Nueva Tarea",
      done: false,
    });
    expect(inputText.value).toBe("");
  });

  test("El formulario con una descripción vacía no debe llamar al onNewTodo", () => {
    render(<TodoAdd onNewTodo={onNewTodoMock} />);
    const submitBtn = screen.getByRole("button");

    fireEvent.click(submitBtn);

    expect(onNewTodoMock).not.toHaveBeenCalled();
  });
});
