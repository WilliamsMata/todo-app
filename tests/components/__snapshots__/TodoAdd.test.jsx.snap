// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoAdd /> Debe de hacer match con el snapshot 1`] = `
<div>
  <form>
    <input
      autocomplete="off"
      class="form-control"
      name="description"
      placeholder="Que hay que hacer?"
      type="text"
      value=""
    />
    <button
      class="btn btn-primary mt-2"
      type="submit"
    >
      Agregar Tarea
    </button>
  </form>
</div>
`;
